### ISSUE TITLE
[BUG] - Docker container does not shutdown cleanly on SIGTERM

### ISSUE THREAD

**Issue by tdiesler** [input-output-hk/cardano-node#2353]
**Internal/External**
*External*  

**Area**
*Docker*
*Shutdown hook*

**Summary**

On bare metal, we can cleanly shutdown a node with SIGINT (i.e. Ctrl+C). The console reports this as `Shutting down..`. In the log we see 

```
Closed db with immutable tip at 9ecca9b01becb4647629d2aac28c661311890609a3a06da8d5a9708f02b1e5d4 at slot 18392924

Shutting down..
```

This is the desired behavior and perhaps most importantly, the node can startup again without significant delay

In a container environment, we may not get SIGINT but SIGTERM instead, see [Gracefully Stopping Docker Containers]( https://www.ctl.io/developers/blog/post/gracefully-stopping-docker-containers/)

As a result, a containerized cardano node does not cleanly close the database and requires full database validation when started again. This may lead to a > 15min delay before the node is operational again

**Steps to reproduce**

```
docker run --rm \
  --name=relay \
  -e NETWORK=mainnet \
  -v shelley-data:/data/db \
  inputoutput/cardano-node

docker stop relay
docker start relay
```

**Expected behavior**

On restart the node should not need to validate the db. The shutdown hook should work for SIGTERM as it does for SIGINT

**Additional context**

You can leave this to me, I'll provide a PR.

CrossRef: https://github.com/tdiesler/nessus-cardano/issues/11

**Comment by robinboening** [comment]
This would be great! 👍

**Comment by tdiesler** [comment]
Thanks. Most of the functionality talked about in [#2360] is already available in [nessusio/cardano](https://hub.docker.com/r/nessusio/cardano) and I only just now feel brave enough to touch the upstream Nix build ;-)

**Comment by robinboening** [comment]
I very quickly skimmed through your repo and will definitely spend some more time there looking into how you've set up some things.

I am currently playing around with a docker based set up for a stake pool. I really just started two days day ago, so I am pretty much exploring the tools and understanding how things are connected.

**Comment by tdiesler** [comment]
There is also this gitbook
https://app.gitbook.com/@tdiesler/s/cardano/

**Comment by tdiesler** [comment]
PR #2375

**Comment by tdiesler** [comment]
This is now part of #2393

**Comment by tdiesler** [comment]
This is now part of #2393

**Comment by tdiesler** [comment]
Merged

**Comment by jbgi** [comment]
Fixed in 952ce084373908f275110f57a43cea338604c4b3.
